pandas/core/internals/blocks.py

```diff
@@ -2585,8 +2585,11 @@ def setitem(self, indexer, value):
         try:
             return super().setitem(indexer, value)
         except (ValueError, TypeError):
+            obj_vals = self.values.astype(object)
```

Comment:
do you *really* need to do this?

if you actually do, then .reshape(1, -1) should just work (or use np.atleast_2d)

Reply:
Yes, this really is needed.

reshape(1, -1) is what is already here

Reply:
can you give an example that specific needs this 2d reshape?

this just is a completely different pattern than *anywhere* else. Why for example would the ObjectBlock construct not simply reshape (if needed); it already knows the dim & the values shape?